code/obj/item/kendo.dm

```diff
@@ -189,51 +185,53 @@
     var/open = 0
     var/shinai = 2
 
-    proc/update_spront(var/mob/user)
+    proc/update_sprite(var/mob/user)
         if(!open)
             src.icon_state = "shinaibag-closed"
             src.item_state = "shinaibag-closed"
         else
-            src.icon_state = "shinaibag-[shinai]"
-            src.item_state = "shinaibag-[shinai]"
+            src.icon_state = "shinaibag-[shinai+src.contents.len]"
+            src.item_state = "shinaibag-[shinai+src.contents.len]"
 
         if(src.loc == user)
             user.update_clothing()
             user.update_inhands()
 
     proc/draw_shinai(var/mob/user)
         if(shinai)
-            var/obj/item/shinai/S = new /obj/item/shinai
-            user.put_in_hand_or_drop(S)
-            S.change_guard(user,user.a_intent)
-            shinai--
-            update_spront(user)
+            if(src.contents.len)
```

Comment:
I think this needs to be before the above `if (shinai)` check? Else once you've taken them all out `shinai` will be `0` and you won't be able to get here to take out the ones you've put back in?

Reply:
oopsie XD fixed